quarkus/runtime/src/main/java/org/keycloak/provider/quarkus/QuarkusPlatform.java

```diff
@@ -122,9 +126,27 @@ public File getTmpDirectory() {
             File tmpDir;
             if (homeDir == null) {
                 // Should happen just in the unit tests
-                homeDir = System.getProperty("java.io.tmpdir");
-                tmpDir = new File(homeDir, "keycloak-quarkus-tmp");
-                tmpDir.mkdir();
+                try {
+                    // Use "tmp" directory in case it points to the "target" directory (which is usually the case with quarkus unit tests)
+                    // Trying to use "target" subdirectory to avoid the situation when separate subdirectory will be created in the "/tmp" for each build and hence "/tmp" directory being swamped with many subdirectories
+                    String tmpDirProp = System.getProperty("java.io.tmpdir");
+                    if (tmpDirProp == null || !tmpDirProp.endsWith("target")) {
+                        // Fallback to "target" inside "user.dir"
+                        String userDirProp = System.getProperty("user.dir");
+                        if (userDirProp != null) {
+                            File userDir = new File(userDirProp, "target");
+                            if (userDir.exists()) {
+                                tmpDirProp = userDir.getAbsolutePath();
+                            }
+                        }
+                    }
+                    // Finally fallback to system tmp directory. Always create dedicated directory for current user
+                    Path path = tmpDirProp != null ? Files.createTempDirectory(new File(tmpDirProp).toPath(), "keycloak-quarkus-tmp", new FileAttribute[]{}) :
+                            Files.createTempDirectory("keycloak-quarkus-tmp", new FileAttribute[]{});
```

Comment:
I don't think you want to pass the `new FileAttribute[]{}` here, do you? I don't think it's needed.

Reply:
Yes, true. Thanks for pointing that